src/invocation/InvocationService.ts

```diff
@@ -686,4 +697,37 @@ export class InvocationService {
     deregisterInvocation(correlationId: number): void {
         this.pending.delete(correlationId);
     }
+
+    checkUrgentInvocationAllowed(invocation: Invocation): void {
+        if (this.connectionRegistry.clientInitializedOnCluster()) {
+            // If the client is initialized on the cluster, that means we
+            // have sent all the schemas to the cluster, even if we are
+            // reconnected to it
+            return;
+        }
+
+        if (!this.schemaService.hasAnySchemas()) {
+            // If there were no Compact schemas to begin with, we don't need
+            // to perform the check below. If the client didn't send a Compact
+            // schema up until this point, the retries or listener registrations
+            // could not send a schema, because if they were, we wouldn't hit
+            // this line.
+            return;
+        }
+
+        // We are not yet initialized on cluster, so the Compact schemas might
+        // not be sent yet. This message contains some serialized classes,
+        // and it is possible that it can also contain Compact serialized data.
+        // In that case, allowing this invocation to go through now could
+        // violate the invariant that the schema must come to cluster before
+        // the data. We will retry this invocation and wait until the client
+        // is initialized on the cluster, which means schemas are replicated
+        // in the cluster.
+        if (invocation.request.isContainsSerializedDataInRequest()) {
+            throw new InvocationMightContainCompactDataError('The invocation' 
+            + invocation + ' might contain Compact serialized '
+            + 'data and it is not safe to invoke it when the client is not '
+            + 'yet initialized on the cluster');
+        }
```

Comment:
```suggestion
        if (invocation.request.isContainsSerializedDataInRequest()) {
            throw new InvocationMightContainCompactDataError('The invocation with correlation id '
            + invocation.request.getCorrelationId() + ' might contain Compact serialized '
            + 'data and it is not safe to invoke it when the client is not '
            + 'yet initialized on the cluster');
        }
```

We can't just put invocation into string like that. It will be something like `[object Object]`. We can at least use the correlation id for now for debugging purposes.

Reply:
Fixed